src/cargo/core/dependency.rs

```diff
@@ -327,14 +339,14 @@ impl Dependency {
         self.inner.matches_id(id)
     }
 
-    pub fn map_source(self, to_replace: &SourceId, replace_with: &SourceId)
+    pub fn map_source(mut self, to_replace: &SourceId, replace_with: &SourceId)
```

Comment:
Nah moving around owned `Rc` by value is super cheap, I don't think I've ever seen it in a profile anywhere. This should arguably just take `&mut self` though.